### PR TITLE
To hell listtransactions

### LINKED ISSUE #74: listtransactions is convoluted
We are trying to do too many things with  `listtransactions`. We need to split it into different purposes.

@edouardparis what do you need from the other side of the UDS ? :)

### PR DESCRIPTION
**Pull request by darosior**
This is based on #90 and fixes #74 .

We split the `listtransactions` command into `listpresignedtransactions` and `listonchaintransactions` commands. Also added my custom script for regtest management to contrib/ and hardcoded config, that's hacky but we can still improve over...

### CODE REVIEW

**Review by edouardparis — APPROVED** [review]
ACK 0bec0bdd5fa4e3b0e2dff6d94ad8e4d79c2ce2ee

File: doc/API.md [hunk 1]
```diff
@@ -132,6 +133,44 @@ vault's state).
...
+| Parameter   | Type         | Description                                                                                     |
+| ----------- | ------------ | ----------------------------------------------------------------------------------------------- |
+| `outpoints` | string array | Vault IDs -- optional, filter the list with the given vault Outpoints                           |
+
+
+### Response
+
+| Field                         | Type                                                 | Description                       |
+| ----------------------------- | ---------------------------------------------------- | --------------------------------- |
+| `onchain_transactions`        | array of [onchain txs](#onchain-txs)                 | Each vault's onchain transactions |
```
Comment (edouardparis) [inline doc/API.md:150]: maybe redundant with the name of the request, `transactions` is enough I think

**Comment by edouardparis** [comment]
ACK df85b7f695d13e03ccc4d16c737c3bce75c871c3

**Comment by darosior** [comment]
ACK 52aa63d -- Re-applying Edouard's ACK after rebasing on top of master, no change

**Pull request merged**